### PR TITLE
Use argparse instead of getops in packaged scans

### PR DESCRIPTION
**Pull request by NoobforAl**
## What does this do ?
1 - use argparse instead of getops ( Task3  :  https://github.com/zaproxy/zaproxy/pull/3566 )
2 -<del> use validators library for validation url </del>( removed! in https://github.com/zaproxy/zaproxy/pull/7213#discussion_r864021766 )

<del>Validators library :  [Validators](https://validators.readthedocs.io/en/latest/)</del>

Signed-off-by: NoobforAl 58856931+NoobforAl@users.noreply.github.com

### CODE REVIEW

**Comment by NoobforAl** [comment]
## Update

1 - fix all LGTM analysis alerts python

**Comment by thc202** [comment]
The indentation/formatting changes should be reverted.

File: docker/zap-api-scan.py [hunk 1]
```diff
@@ -49,14 +49,15 @@
...
-import getopt
 import json
 import logging
 import os
 import os.path
 import subprocess
 import sys
 import time
+import argparse
+from validators import url
```
[the remark this answers is not in the document]
Comment (NoobforAl) [reply docker/zap-api-scan.py:60]: OK!(https://github.com/zaproxy/zaproxy/pull/7213#issuecomment-1116097846) I will fix it!

validators ?
why ?
In the past code, checking was very simple! [code](https://github.com/zaproxy/zaproxy/blob/main/docker/zap-full-scan.py#L217)

[the remark this answers is not in the document]
Comment (thc202) [reply docker/zap-api-scan.py:60]: To have ZAP do the actual validations, which might be more lenient that what it's being done by `validators`.

[the remark this answers is not in the document]
Comment (NoobforAl) [reply docker/zap-api-scan.py:60]: I got it!
I will fix it!

[the remark this answers is not in the document]
Comment (NoobforAl) [reply docker/zap-api-scan.py]: **Done!**

[the remark this answers is not in the document]
Comment (thc202) [reply docker/zap-api-scan.py]: There are still some indentation/formatting changes.

File: docker/zap-api-scan.py [hunk 2]
```diff
@@ -83,74 +83,87 @@ class NoUrlsException(Exception):
...
+DESCRIPTION = '''                                                                                                                             
+d88888P  d888b8b  ?88,.d88b,  88bd88b d8888b ?88,  88P?88   d8P 
+   d8P' d8P' ?88  `?88'  ?88  88P'  `d8P' ?88 `?8bd8P'd88   88  
+ d8P'   88b  ,88b   88b  d8P d88     88b  d88 d8P?8b, ?8(  d88  
+d88888P'`?88P'`88b  888888P'd88'     `?8888P'd8P' `?8b`?88P'?8b 
+                    88P'                                     )88
+                   d88                                      ,d8P
+                   ?8P                                   `?888P'
+
+
```
[the remark this answers is not in the document]
Comment (NoobforAl) [reply docker/zap-api-scan.py]: removed !

File: docker/zap-api-scan.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (NoobforAl) [reply docker/zap-api-scan.py]: I think it's now ok !

**Comment by NoobforAl** [comment]
Hi @thc202, 
Should me fix this  ( This branch has conflicts that must be resolved )?
or those with write access this repository ?
[ I read [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) but i want to sure! ]

**Comment by NoobforAl** [comment]
I think it done!
Thanks! @kingthorin

**Comment by NoobforAl** [comment]
HI,
Excuses me if this change have problem, I fix it!

**Pull request closed without merging**